tests/installation/test_executor.py

```diff
@@ -739,6 +787,60 @@ def test_executor_should_write_pep610_url_references_for_urls(
     verify_installed_distribution(tmp_venv, package, expected_url_reference)
 
 
+@pytest.mark.parametrize("cached", [False, True])
+def test_executor_should_write_pep610_url_references_for_non_wheel_urls(
+    tmp_venv: VirtualEnv,
+    pool: RepositoryPool,
+    config: Config,
+    io: BufferedIO,
+    mock_file_downloads: None,
+    mocker: MockerFixture,
+    fixture_dir: FixtureDirGetter,
+    cached: bool,
+):
+    if cached:
+        link_cached = fixture_dir("distributions") / "demo-0.1.0.tar.gz"
+        original_func = Chef.get_cached_archive_for_link
+
+        def mock_get_cached_archive_for_link(self: Self, link: Link, strict: bool):
+            if link.filename == "demo-0.1.0.tar.gz":
+                return link_cached
+            else:
+                return original_func(self, link, strict)
```

Comment:
That doesn't work as expected. The mock is always called with the tar.gz. It should be something like:

```suggestion
            if strict:
                return cached_sdist
            else:
                return cached_wheel
```

Reply:
You are right. I was about to write a reasoning why it would not but then it dawned on me.

Originally i thought since its a full sdist install, it needs further dependencies like 'poetry-core' during the `chef._prepare` call. This causes additional calls to `mock_get_cached_archive_for_link` for those deps which need to be handled.

I can also see two other ways work:

1. Mocking away all other required deps here with specific filename matches (in my view brittle and hard to maintain)
2. Also mocking away the `chef._prepare` the call. Then it would similiarily become less integration test and more proper unit test, but the `coverage` of the testcase would go down. (Also did not think of that one yesterday).

But if you are fine with mocking that specific case and avoiding the actual build process of the wheel, its good enough for me.

Reply:
Mocking `chef._prepare` should be fine since preparing sdists is tested in `test_chef.py`. If we mock it, we can add an additional check that `chef._prepare` is only called if artifacts are not cached and is not called if there is a cached wheel.

Reply:
I improved the tests and made them more comprehensive. I hope its still readable / understandable.